src/Store/Reducers/animationReducer.js

```diff
@@ -0,0 +1,34 @@
+const initialState = [];
+
+function animationReducer(state = initialState, action) {
+  let nextState;
+  switch (action.type) {
+    case 'SAVE_ANIMATION':
+      // If the saved animation previously had another title, remove the animation with this old title
+      let oldAnimationIndex = state.findIndex(item => item.title === action.value.oldTitle);
+
+      if (oldAnimationIndex !== -1) nextState = state.filter((item, index) => index !== oldAnimationIndex);
+      else nextState = [...state];
+
+      // If the animation is already saved, replace it; otherwise add it
+      let animationIndex = state.findIndex(item => item.title === action.value.title);
+      if (animationIndex !== -1) {
+        nextState[animationIndex] = action.value;
```

Comment:
All these Array manipulations make me wonder if we should have an Array or an Object in the store. If we had an object with the form { drillName: drill }, all these accesses would be much more easier to do (and less complex)

I just don't know (but the internet probaably does) if it is a good idea to have both arrays and objects in our root reducer